snuba_sdk/expressions.py

```diff
@@ -116,3 +126,41 @@ def validate(self) -> None:
                 raise InvalidExpression(
                     f"parameter '{param}' of function {self.function} is an invalid type"
                 )
+
+    def __eq__(self, other: object) -> bool:
+        # Don't use the alias to compare equality
```

Comment:
I couldn't think of a scenario where we would want to include the alias in the equality. Since the semantic equality seems to be more used, I made that the default. If a strong equality is needed I can add it back.